File: BisectionMethod.cpp
#include <iostream>
#include <vector>

// this is test for git ok ??
using namespace std;

class BisectionMethod
{
    // for function     ax^2 + bx + c = 0;
    private :    
    int a,b,c;
    double root;
    int NoOfIterations = 100;
    double range_1,range_2,tempc;
    
    
    public:
    BisectionMethod(){}

    BisectionMethod(int a,int b, int c): a(a),b(b),c(c), root(-1.0){range_1=6; range_2= 7; tempc = 0;}

    double FunctionVal(double x)
    {
        
        return (a*x*x + b*x + c);
    }

    bool CalculateRoot()
    {   
      
        if((FunctionVal(range_1) * FunctionVal(range_2) )> 0 ) return false;
        while(NoOfIterations>0)
        {
        
              
        tempc= (range_1 + range_2)/2;
        if((FunctionVal(tempc)>0 )&& (FunctionVal(range_1)<0)) {range_2 = tempc;} else {range_1 = tempc;}
         
        NoOfIterations--;
            
        }
      
        root = tempc;
        return true;
    }

    void displayRoot()
    {
            cout << " Positive Root : "<< root<<endl;
    }


};

class Interpolation
{
    private:
    vector<vector <float>> Val;

    public:
    Interpolation(){}
    Interpolation(vector<vector <float>> Value) : Val(Value){}

    bool CreateFxn()
    {

        return false;
    }
    
    double ValueAt(int x)
    {
        return 0.0;
    }
        


};

int main ()

{
    int a =1, b=-7, c = 2;
    BisectionMethod SqFxn(a,b,c);
    if(SqFxn.CalculateRoot()) SqFxn.displayRoot(); else { cout << "Error !! "<<endl;}

    vector<vector<float>> Val = {{2,3},{3,6},{4,12},{5,24}};
    Interpolation Intp;
    Intp.CreateFxn();
    cout << " Val : " << Intp.ValueAt(3);


} //cout<<"Before ::" <<"range_1 : "<<range_1<<" range_2: "<<range_2<< " Tempc: " << tempc << " Fxn val : "<< FunctionVal(tempc)*FunctionVal(range_1)<<endl;
    


#include <iostream>
int main (int argc, char* argv[])
{
    std::cout<<"Hello World"<<std::endl;
    return 0;
}


class Person{
private:
    int a,b,c,d,e;
public:

Person(int a,int b,int c,int d, int e):a(a),b(b),c(c),d(d),e(e){}


};

File: BisectionMethod.py
class BisectionMethod:
    def __init__(self, a, b, c, rn1, rn2,toleration):
        self.a = a
        self.b = b 
        self.c = c 
        self.range_1 = rn1
        self.range_2 = rn2
        self.numberofiterations = 1000
        self.tempc = 0
        self.root = -1
        self.tolerance = toleration
        
    def FunctionVal(self, x):
        return (self.a*(x**2) + self.b * x + self.c)
    
    def CalculateRootIteratively(self):
        if ((self.FunctionVal(self.range_1) * self.FunctionVal(self.range_2)) > 0):
            return False

        while (self.numberofiterations>0):            
            self.tempc = (self.range_1 + self.range_2) / 2
            if((self.tempc*self.a)>0):
                self.a = self.tempc 
            else:
                self.b = self.tempc    
            self.numberofiterations -= 1                  
        self.root = self.tempc
        return True
    
    def displayroot(self):
        print(f"Positive Root of Polynomial is : {self.root:.4f}")


# y = x^2 -5x + 6

BisecVar = BisectionMethod(1, -5, 6, 1.5, 2.5,0.0001)
if BisecVar.CalculateRootIteratively():
    BisecVar.displayroot()